types/index.d.ts

```diff
@@ -0,0 +1,61 @@
+declare module 'machina' {
+  interface BehavioralFsmInstance<S> {
+    __machina__?: {
+      [namespace: string]: {
+        targetReplayState: S;
+        state: S;
+        priorState: S;
+        priorAction: string;
+        currentAction: string;
+        currentActionArgs: any[];
+        inputQueue: any[];
+        inExitHandler: boolean;
+      }
+    }
+  }
+
+  type TransitionHandler<T, S> = BehavioralFsmHandler<T, S> | S;
+
+  interface StateTransitions<T, S> {
+    [state: string]: {
+      '*'?: TransitionHandler<T, S>;
+      _onEnter?: TransitionHandler<T, S>;
+      _onExit?: TransitionHandler<T, S>;
+      timeout?: TransitionHandler<T, S>;
```

Comment:
While "timeout" could be an event that a state handles, it's not any sort of special handler like *, _onEnter, or _onExit, so it's covered by line 26:
`[handler: string]: TransitionHandler<T, S> | undefined;`

Ironically, I think this mistake is probably traceable back to me.